src/modules/kvs/kvs.c

```diff
@@ -1053,6 +1072,7 @@ static void kvstxn_apply (kvstxn_t *kt)
         }
         setroot (ctx, root, kvstxn_get_newroot_ref (kt), root->seq + 1);
         setroot_event_send (ctx, root, names, kvstxn_get_keys (kt));
+        start_sync_timer (ctx, root);
```

Comment:
By chance is the timer re-armed here by the sync (empty commit) itself?  I'm noting that even when the system should be idle I'm getting repeated syncs as though the timer were periodic.

Reply:
hmm that makes sense, although I'm not sure how this isn't being caught by tests.

Reply:
ahhh, my tests only ensure that the rootref hasn't changed, which doesn't mean syncs won't happen.  I assume you're putting in printfs or something and notice the sync callback being called more times than it should?

We need another flag to ensure this doesn't happen, one for the sync timer and one for a sync being active.

But b/c of the racy thing I mentioned above, (a commit could complete before the `sync_continuation()` is called), I think this would involve more hackery than I'd like.  Lets talk on the coffee time about this.

This has certainly exceeded the complexity I originally imagined this PR to be, I'm not sure if you had other ideas as well.

Reply:
OK.  No coffee time today though (it's tuesday).  Did you want to chat now?  LMK when you're available.

They only race I can imagine is maybe if the sync commit took longer than a concurrent timer.

I didn't entirely understand  how the empty commit is handled, but it would seem like it must be publishing the root if it's also starting the timer since that's right next to it in the code?  It probably shouldn't be doing that either, so maybe it needs to short circuit out of there earlier?